src/reporting/Market.sol

```diff
@@ -0,0 +1,593 @@
+        ReportingWindow _newReportingWindow = getBranch().getReportingWindowForForkEndTime();
+        return migrateReportingWindow(_newReportingWindow);
+    }
+
+    function migrateReportingWindow(ReportingWindow _newReportingWindow) private afterInitialized returns (bool) {
+        _newReportingWindow.migrateMarketInFromSibling();
+        reportingWindow.removeMarket();
+        reportingWindow = _newReportingWindow;
+        return true;
+    }
+
+    function updateTentativeWinningPayoutDistributionHash(bytes32 _payoutDistributionHash) public returns (bool) {
+        ReportingToken _reportingToken = reportingTokens[_payoutDistributionHash];
+        require(address(_reportingToken) != NULL_ADDRESS);
+
+        ReportingToken _tentativeWinningReportingToken = reportingTokens[tentativeWinningPayoutDistributionHash];
+        if (address(_tentativeWinningReportingToken) == NULL_ADDRESS) {
+            tentativeWinningPayoutDistributionHash = _payoutDistributionHash;
+            _tentativeWinningReportingToken = _reportingToken;
+        }
+        if (_reportingToken.totalSupply() > _tentativeWinningReportingToken.totalSupply()) {
+            tentativeWinningPayoutDistributionHash = _payoutDistributionHash;
+        }
+        return true;
+    }
+
+    function tryFinalize() public returns (bool) {
+        require(tentativeWinningPayoutDistributionHash != bytes32(0));
+
+        if (isFinalized()) {
+            return true;
+        }
+
+        if (getReportingState() != ReportingState.AWAITING_FINALIZATION) {
+            return false;
+        }
+
+        finalPayoutDistributionHash = tentativeWinningPayoutDistributionHash;
+        finalizationTime = block.timestamp;
+        transferIncorrectDisputeBondsToWinningReportingToken();
+        reportingWindow.updateMarketPhase();
+        return true;
+
+        // FIXME: when the market is finalized, we need to add `reportingTokens[finalPayoutDistributionHash].totalSupply()` to the reporting window.  This is necessary for fee collection which is a cross-market operation.
+        // TODO: figure out how to make it so fee distribution is delayed until all markets have been finalized; we can enforce it contract side and let the UI deal with the actual work
+        // FIXME: if finalPayoutDistributionHash != getIdentityDistributionId(), pay back validity bond holder
+        // FIXME: if finalPayoutDistributionHash == getIdentityDistributionId(), transfer validity bond to reportingWindow (reporter fee pot)
+        // FIXME: if automated report is wrong, transfer automated report bond to reportingWindow
+        // FIXME: if automated report is right, transfer automated report bond to market creator
+        // FIXME: handle markets that get 0 reports during their scheduled reporting window
+    }
+
+    function canFinalizeAutomated() private returns (bool) {
+        if (!automatedReportReceived) {
+            return false;
+        }
+        return true;
+    }
```

Comment:
This may have been a pre-existig bug, but this needs to check more than just `automatedReportReceived`.  At the least, it needs to verify no dispute bond was placed and it is after the dispute period.  As with several other methods in this contract, consider removing it and replacing its callsites with `getReportingState`.

As you may be catching on to, now that we have a consolidated "Reporting State" system I'm a big fan of moving more things over to it rather than having all of these one-off methods that do some subset of state checks.  There are some minor performance advantages to checking only a subset of the state, but I think we should opt for cleaner code at the moment and performance optimize later if necessary.